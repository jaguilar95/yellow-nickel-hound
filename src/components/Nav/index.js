import React from "react";

function Nav(props) {
  const { sections = [] } = props;
  return (
    <header className="flex-row px-1">
      <h2>
        <a href="/">j.aguilar</a>
      </h2>
      <nav>
        <ul className="flex-row">
          {sections.map((section) => (
            <li className="mx-1" key={section.name}>
              {section.name}
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
}

export default Nav;
