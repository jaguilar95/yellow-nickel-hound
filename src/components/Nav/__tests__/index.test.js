import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

const sections = [
  { name: "portraits", description: "Portraits of people in my life" },
];

afterEach(cleanup);

describe("Nav component", () => {
  // baseline test
  it("renders", () => {
    render(<Nav sections={sections} />);
  });

  // snapshot test
  it("nav render matches snapshot", () => {
    const { asFragment } = render(<Nav sections={sections} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
