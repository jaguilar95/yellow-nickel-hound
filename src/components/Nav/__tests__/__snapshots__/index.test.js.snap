// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component nav render matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h2>
      <a
        href="/"
      >
        j.aguilar
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-1"
        >
          portraits
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
